src/packages/version-check/tests/version-check.test.ts

```diff
@@ -0,0 +1,737 @@
+      vc = new VersionCheck(testVersion, testConfig, testLogger);
+
+      assert.notStrictEqual(
+        vc._config.lastVersionLogged,
+        testConfig.latestVersion,
+        "latestVersion and lastVersionLogged is the same before logging."
+      );
+
+      vc.log();
+
+      assert.strictEqual(
+        vc._config.lastVersionLogged,
+        testConfig.latestVersion,
+        "lastVersionLogged was not successfully set after logging version message."
+      );
+    });
+    it("sets lastNotification", async () => {
+      vc = new VersionCheck(testVersion, testConfig, testLogger);
+
+      assert.notStrictEqual(
+        vc._config.lastVersionLogged,
+        testConfig.latestVersion,
+        "latestVersion and lastVersionLogged is the same before logging."
+      );
+
+      const firstNotification = vc._config.lastNotification;
+
+      vc.log();
+
+      const secondNotification = vc._config.lastNotification;
+
+      assert.strictEqual(
+        secondNotification > firstNotification,
+        true,
+        "lastVersionLogged was not successfully set after logging version message."
+      );
+    });
+    it("only logs the latestVersion one time", () => {
+      vc = new VersionCheck(testVersion, testConfig, testLogger);
+
+      vc.log();
+
+      assert.strictEqual(
+        message.length > 0,
+        true,
+        "Log did not log the first time for this version."
+      );
+      message = "";
+      vc.log();
+
+      assert.strictEqual(message, "", "logged the same version message twice.");
+    });
+
+    it("message contains the upgradeType", () => {
+      vc = new VersionCheck(testVersion, testConfig, testLogger);
+      vc.log();
+
+      const upgradeType = semverUpgradeType(
+        vc._currentVersion,
+        vc._config.latestVersion
+      );
+
+      assert.strictEqual(
+        upgradeType && message.indexOf(upgradeType) >= 0,
+        true,
+        "Message does not contain the upgradeType."
+      );
+    });
+
+    it("message contains the packageName", () => {
+      vc = new VersionCheck(testVersion, testConfig, testLogger);
+      vc.log();
+
+      assert.strictEqual(
+        message.indexOf(vc._config.packageName) >= 0,
+        true,
+        "Message does not contain the upgradeType."
+      );
+    });
+    it("message contains the currentVersion", () => {
+      vc = new VersionCheck(testVersion, testConfig, testLogger);
+      vc.log();
+
+      assert.strictEqual(
+        message.indexOf(vc._currentVersion) >= 0,
+        true,
+        "Message does not contain the upgradeType."
+      );
+    });
+    it("message contains the latestVersion", () => {
+      vc = new VersionCheck(testVersion, testConfig, testLogger);
+      vc.log();
+
+      assert.strictEqual(
+        message.indexOf(vc._config.latestVersion) >= 0,
+        true,
+        "Message does not contain the upgradeType."
+      );
+    });
+    it("process.stdout.columns === -1", () => {
```

Comment:
what is the desired behaviour in this case? It'd be nice to document that in the test title

Reply:
oh I have no idea. I inherited all the banner message code and left it as is. I just tested the functional parts.

I would have just gone with the cli Message but I'm all for whatever banner y'all would like to see.